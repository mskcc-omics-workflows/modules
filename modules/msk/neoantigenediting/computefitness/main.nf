process NEOANTIGENEDITING_COMPUTEFITNESS {
    tag "$meta.id"
    label 'process_medium'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/neoantigenediting:1.2':
        'docker.io/mskcc/neoantigenediting:1.2' }"

    input:
    tuple val(meta),  path(patient_data), path(alignment_file)

    output:
    tuple val(meta), path("*_annotated.json")               , emit: annotated_output
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    python3 /usr/bin/compute_fitness.py \\
        --alignment ${alignment_file} \\
        --input ${patient_data}


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        neoantigenEditing: \$NEOANTIGEN_EDITING_TAG
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """

    touch patient_data_annotated.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        neoantigenEditing: \$NEOANTIGEN_EDITING_TAG
    END_VERSIONS
    """
}
