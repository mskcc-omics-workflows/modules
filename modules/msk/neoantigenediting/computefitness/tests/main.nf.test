nextflow_process {

    name "Test Process NEOANTIGENEDITING_COMPUTEFITNESS"
    script "../main.nf"
    process "NEOANTIGENEDITING_COMPUTEFITNESS"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "neoantigenediting"
    tag "neoantigenediting/computefitness"

    test("neoantigenediting_computefitness - json") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map,
                    file(params.test_data_mskcc['neoantigen']['patient_data'], checkIfExists: true),
                    file(params.test_data_mskcc['neoantigen']['iedb_alignments'], checkIfExists: true)
                    ]
                input[0] = [
                    [ id:'test', single_end:false ], // meta map,
                    file(params.test_data_mskcc['neoantigen']['patient_data'], checkIfExists: true),
                    file(params.test_data_mskcc['neoantigen']['iedb_alignments'], checkIfExists: true)
                    ]
                

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions,file(process.out.annotated_output[0][1]).name).match() }

            )
        }

    }

    test("neoantigenediting_computefitness - json - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map,
                    file('patient_data'),
                    file('alignment_file')
                    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions,process.out).match() }

            )
        }

    }


}
