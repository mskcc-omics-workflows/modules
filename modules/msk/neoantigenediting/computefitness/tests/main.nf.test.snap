{
    "neoantigenediting_computefitness - json - stub": {
        "content": [
            [
                "versions.yml:md5,d615e02977a234e01fe78e023a51362a"
            ],
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "patient_data_annotated.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d615e02977a234e01fe78e023a51362a"
                ],
                "annotated_output": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "patient_data_annotated.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d615e02977a234e01fe78e023a51362a"
                ]
            }
        ],
        "timestamp": "2024-06-19T14:54:10.949451"
    },
    "neoantigenediting_computefitness - json": {
        "content": [
            [
                "versions.yml:md5,d615e02977a234e01fe78e023a51362a"
            ],
            "3_OLTS_primary_tumor_annotated.json"
        ],
        "timestamp": "2024-06-19T14:54:04.735114"
    }
}